hack/tools/tilt-prepare/main.go

```diff
@@ -70,17 +70,23 @@ var (
 	rootPath      string
 	tiltBuildPath string
 
-	toolsFlag            = pflag.StringSlice("tools", []string{}, "list of tools to be created; each value should correspond to a make target")
-	certManagerFlag      = pflag.Bool("cert-manager", false, "prepare cert-manager")
-	kustomizeBuildsFlag  = pflag.StringSlice("kustomize-builds", []string{}, "list of kustomize build to be run; each value should be in the form name:path")
-	providersBuildsFlag  = pflag.StringSlice("providers", []string{}, "list of providers to be installed; each value should be in the form name:path")
-	allowK8SContextsFlag = pflag.StringSlice("allow-k8s-contexts", []string{}, "Specifies that Tilt is allowed to run against the specified k8s context name; Kind is automatically allowed")
 	tiltSettingsFileFlag = pflag.String("tilt-settings-file", "./tilt-settings.yaml", "Path to a tilt-settings.(json|yaml) file")
 )
 
 type tiltSettings struct {
-	Debug     map[string]debugConfig `json:"debug,omitempty"`
-	ExtraArgs map[string]extraArgs   `json:"extra_args,omitempty"`
+	Debug               map[string]debugConfig `json:"debug,omitempty"`
+	ExtraArgs           map[string]extraArgs   `json:"extra_args,omitempty"`
+	CertManager         bool                   `json:"deploy_cert_manager,omitempty"`
+	DeployObservability []string               `json:"deploy_observability,omitempty"`
+	ProviderBuilds      []string               `json:"enable_providers,omitempty"`
+	Tools               []string               `json:"tools,omitempty"`
+	K8sContext          []string               `json:"allowed_contexts,omitempty"`
+	ProviderRepos       []string               `json:"provider_repos,omitempty"`
```

Comment:
```suggestion
	Debug               map[string]debugConfig `json:"debug,omitempty"`
	ExtraArgs           map[string]extraArgs   `json:"extra_args,omitempty"`
	DeployCertManager         bool                   `json:"deploy_cert_manager,omitempty"`
	DeployObservability []string               `json:"deploy_observability,omitempty"`
	EnableProviders      []string               `json:"enable_providers,omitempty"`
	ProviderRepos       []string               `json:"provider_repos,omitempty"`
	Tools               []string               `json:"tools,omitempty"`
	AllowedContexts          []string               `json:"allowed_contexts,omitempty"`
```
Let's use the field names which match the JSON names, otherwise it's really hard to map